library/modules/bigip_gtm_dns_listener.py

```diff
@@ -0,0 +1,943 @@
+        See the documentation for an example.
+    type: list
+    elements: str
+  advertise:
+    description:
+      - Specifies whether this listener's address is advertised to surrounding routers.
+    type: bool
+  auto_lasthop:
+    description:
+      - Specifies whether to automatically map last hop for pools or not.
+    type: str
+  last_hop_pool:
+    description:
+      - Specifies the name of the last hop pool that you want the listener to use to direct reply traffic to the last hop router.
+    type: str
+  fallback_persistence:
+    description:
+      - Specifies a fallback persistence profile for the listener to use when the default persistence profile is not available.
+    type: str
+  ip_protocol:
+    description:
+      - Specifies the protocol on which this listener receives network traffic.
+    type: str
+  partition:
+    description:
+      - Device partition to manage resources on.
+    type: str
+    default: Common
+  state:
+    description:
+      - DNS Listener state.
+      - When C(present), ensures that the pool is created and enabled.
+      - When C(absent), ensures that the pool is removed from the system.
+      - When C(enabled) or C(disabled), ensures that the pool is enabled or disabled respectively) on the remote device.
+    type: str
+    choices:
+      - present
+      - absent
+      - enabled
+      - disabled
+    default: present
+extends_documentation_fragment: f5networks.f5_modules.f5
+author:
+  - Andrey Kashcheev (@andreykashcheev)
+'''
+
+EXAMPLES = r'''
+- name: 'Create Vlans'
+  bigip_vlan:
+    delegate_to: localhost
+    loop:
+      - external
+      - internal
+    name: '{{ item }}'
+    provider:
+      password: secret
+      server: lb.mydomain.com
+      user: admin
+- name: 'Create vlans'
+  bigip_irule:
+    content: 'when DNS_REQUEST {}'
+    delegate_to: localhost
+    loop:
+      - irule1
+      - irule2
+      - irule3
+      - irule4
+    module: ltm
+    name: '{{ item }}'
+    provider:
+      password: secret
+      server: lb.mydomain.com
+      user: admin
+- name: 'Create iRules'
+  bigip_gtm_dns_listener:
+    address: '192.0.1.0'
+    advertise: false
+    auto_lasthop: default
+    description: 'this is a test DNS listener'
+    enabled_vlans:
+      - /Common/external
+    ip_protocol: tcp
+    irules:
+      - /Common/irule1
+    mask: '255.255.255.0'
+    name: test-dns-listener
+    port: 30025
+    provider:
+      password: secret
+      server: lb.mydomain.com
+      user: admin
+    source_port: preserve
+    state: present
+    translate_address: true
+    translate_port: true
+  delegate_to: localhost
+- name: 'Disable a DNS Listener'
+  bigip_gtm_dns_listener:
+    address: '192.0.1.0'
+    state: disabled
+    name: test-dns-listener
+    provider:
+      password: secret
+      server: lb.mydomain.com
+      user: admin
+  delegate_to: localhost
```

Comment:
space these out each task separated by a space, also not needed to include tasks from other modules

Reply:
This comment is addressed now